docs/getting-started/rulesets.md

```diff
@@ -249,6 +249,51 @@ The example above will run all of the rules defined in the `spectral:oas` rulese
 
 - [Rules relevant to OpenAPI v2 and v3](../reference/openapi-rules.md)
 
+## Selectively silencing some results
+
+From time to time, you want to ignore some specific results without turning off 
+the rule entirely. This may happen, for instance, when working with legacy APIs.
+
+The ruleset can be extended to define such exceptions through the optional `except` property.
+
+`except` describes a map of locations (expressed as Json paths) and rules that should be ignored.
+
+Locations be can either described as relatively to the ruleset or absolute paths.
+
+```yaml
+extends: spectral:oas
+
+except:
+  "subfolder/one.yaml#"
+    - oas3-api-servers
+  "/tmp/docs/one.yaml#/info":
+    - info-contact
+    - info-description
+```
+
+If you want to pass your document to Spectral through `stdin`, the ruleset will 
+have to be expressed slightly differently, using a predefined `<STDIN>` pattern as
+the location.
+
+For instance, running the following command will honor the defined exceptions...
+
+```bash
+$ cat /path/to/the/document | spectral lint -r /path/to/ruleset.yaml
+```
+
+...provided the ruleset is defined this way:
+
+```yaml
+extends: spectral:oas
+
+except:
+  "<STDIN>#":
```

Comment:
I don't think we need to mess around with this. I think putting something in the ruleset which demands knowledge of exactly how it's being called is going to be more confusion than its worth.

I would expect to see people writing `#/info`, or `models/awful.json` to make it shut up about issues in info, or a particularly awful model. If that info ignore just so happens to quieten up two different infos im not sure people will be that upset. 

Let's try and avoid overthinking things, and leave room for more improvements later if we get feedback that they're required. I don't think entrance files are the biggest concern for people, and if they are a more specific exception can be written, so this will possibly not come up for a long time - if ever. 🎲

Reply:
@philsturgeon Thanks for the detailed feedback!

The current implementation was designed to help make one of the initial requirements happen (from https://github.com/stoplightio/spectral/issues/747#issuecomment-568494226)

![image](https://user-images.githubusercontent.com/92363/75855329-4b98df00-5df2-11ea-8bb0-d367aa6cdb2a.png)

As such, the current syntax expects a file and a non ambiguous path within it that should be ignored.

Of course, the model can be extended to support more "handwritten" scenarios:

- `models/awful.json: [ rule1, rule2]`: Would ignore rule1 and rule2 results wheverever in the file
- `#/info": [rule1, rule2]`: Would ignore rule1 and rule2 results in any processed document provided they originate from a path that's `#/info` **or below (eg. `#/info/sub/path/1`)**. This would be in line with @P0lip's [comment](https://github.com/stoplightio/spectral/pull/939#discussion_r371174774)

Would that fit you better?

Reply:
Ok if that’s the goal just ignore STDIN entirely and support files.

Reply:
@philsturgeon Dropped. STDIN and laxer syntax will be tracked through separate issues to keep that PR on scope.